pkg/servicemesh/ossmcommon/operations_unit_test.go

```diff
@@ -0,0 +1,72 @@
+package ossmcommon_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/servicemesh/ossmcommon"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var _ = Describe("SetGatewayName", func() {
+	var (
+		tempDir                             string
+		dscispec                            *dsci.DSCInitializationSpec
+		KfnotebookControllerServiceMeshPath string
+	)
+
+	BeforeEach(func() {
+		var err error
+
+		tempDir, err = os.MkdirTemp("", "ossm-test")
```

Comment:
Starting from Go 1.15, the standard testing package includes [`T.TempDir()`](https://pkg.go.dev/testing#T.TempDir), perhaps worth giving a try, especially that it's fully supported by Ginkgo `GinkgoT().TempDir()`. You will at least save few lines of code for cleanup :)